pkg/test/vpc_client/bastion.go

```diff
@@ -111,38 +115,57 @@ func (vpc *VPC) PrepareBastionProxy(zone string, cidrBlock string, keypairName s
 
 	insts, err := vpc.AWSClient.ListInstances([]string{}, filters...)
 	if err != nil {
-		return nil, err
+		return nil, "", "", err
 	}
 	if len(insts) == 0 {
 		log.LogInfo("Didn't found an existing bastion, going to launch one")
-		if cidrBlock == "" {
-			cidrBlock = CON.RouteDestinationCidrBlock
-		}
-		_, _, err = net.ParseCIDR(cidrBlock)
-		if err != nil {
-			log.LogError("CIDR IP address format is invalid")
-			return nil, err
-		}
 
-		userData := fmt.Sprintf(`#!/bin/bash
+		userData := `#!/bin/bash
 		yum update -y
-		yum install -y squid
+		sudo dnf install squid -y
 		cd /etc/squid/
 		sudo mv ./squid.conf ./squid.conf.bak
 		sudo touch squid.conf
 		echo http_port 3128 >> /etc/squid/squid.conf
-		echo acl allowed_ips src %s >> /etc/squid/squid.conf
-		echo http_access allow allowed_ips >> /etc/squid/squid.conf
+		echo auth_param basic program /usr/lib64/squid/basic_ncsa_auth /etc/squid/passwords >> /etc/squid/squid.conf
+		echo auth_param basic realm Squid Proxy Server >> /etc/squid/squid.conf
+		echo acl authenticated proxy_auth REQUIRED >> /etc/squid/squid.conf
+		echo http_access allow authenticated >> /etc/squid/squid.conf
 		echo http_access deny all >> /etc/squid/squid.conf
 		systemctl start squid
-		systemctl enable squid`, cidrBlock)
+		systemctl enable squid`
 
 		encodeUserData := base64.StdEncoding.EncodeToString([]byte(userData))
-		return vpc.LaunchBastion("", zone, encodeUserData, keypairName, privateKeyPath)
+		instance, err := vpc.LaunchBastion("", zone, encodeUserData, keypairName, privateKeyPath)
+		if err != nil {
+			log.LogError("Launch bastion failed")
+		}
+
+		username := utils.RandomLabel(5)
+		password := utils.RandomLabel(5)
 
+		hashedPassword, err := generateBcryptPassword(password)
+		if err != nil {
+			return nil, "", "", err
+		}
+
+		localFilePath := "./tmp/passwords"
+		err = writePasswordToFile(username, hashedPassword, localFilePath)
+		if err != nil {
+			return nil, "", "", err
+		}
+
+		remoteFilePath := "/etc/squid/passwords"
+		err = uploadFileToBastion(*instance.PublicIpAddress, "22", "ec2-user", privateKeyPath,
+			keypairName, localFilePath, remoteFilePath)
+
+		if err != nil {
+			return nil, "", "", err
+		}
+		return instance, username, password, nil
 	}
 	log.LogInfo("Found existing bastion: %s", *insts[0].InstanceId)
-	return &insts[0], nil
+	return &insts[0], "", "", nil
```

Comment:
Let's return the proxy url here rather than instance.

Reply:
Updated according to comments. Proxy url will be returned currently.